src/miniscript/mod.rs

```diff
@@ -1408,6 +1408,39 @@ mod tests {
         ms.translate_pk(&mut t).unwrap_err();
     }
 
+    #[test]
+    fn mixed_timelocks() {
```

Comment:
I am surprised that this did not exist before. Anyways more tests are always welcome.

Reply:
Possibly one existed somewhere. I grepped for the timelock-specific errors and found nothing. But there are a ton of tests that just assert `.unwrap_err()` and don't check more specifically.

And I'm pretty confident that there is no test that covers all three separate checks (in `Miniscript`, `Semantic` and `Concrete`).